contracts/crypto-verify/src/contract.rs

```diff
@@ -553,17 +553,19 @@ mod tests {
 
         let mut messages: Vec<Binary> = [ED25519_MESSAGE_HEX, ED25519_MESSAGE2_HEX]
             .iter()
-            .map(|m| Binary(hex::decode(m).unwrap()))
+            .map(|m| Binary::new(hex::decode(m).unwrap()))
             .collect();
         // Alter one of the messages
-        messages[0].0[0] ^= 0x01;
+        let mut b: Vec<_> = messages.pop().unwrap().into();
+        b[0] ^= 0x01;
+        messages.push(Binary::new(b));
```

Comment:
This changes the test from the first to the last message. Not sure if it matters though.

Isn't there a way to get a mut ref without removing the element from the vector? Something like

```rust
let element = &mut messages[0];
// alter element
```

or https://doc.rust-lang.org/std/primitive.slice.html#method.get_mut

Reply:
That would give us a `&mut Binary`, but the problematic part is that `// alter element`:
`&mut Binary` does not allow us to modify the bytes directly (there is no `^=` or `^` impl for `Binary`).
It also does not allow us to get a `&mut [u8]` that we could then modify.
To allow that, we have to implement `DerefMut`.